### PR TITLE
Added API configuration page/endpoint (MVC)

### PR DESCRIPTION
**Pull request by tbarsballe**
API Configuration page located at `geoserver/api`
Lists all possible api requests, and the associated methods.

@ischneider I have also incorporated your IconController fix.

### CODE REVIEW

File: geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<body>
+<h2>Composer API Configuration</h2>
+<#if model["requests"]?size != 0>
+<ul>
+<#foreach request in model["requests"]>
+<li>${request["method"]} <a href="${model["baseUrl"]}${request["pattern"]}">${request["pattern"]}</a></li>
+</#foreach>
+</ul>
+<#else>
+<p>There are no REST extensions installed.  If you expected some, please verify your installation (did you restart the server?).</p>
```
Comment (ischneider) [inline geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl]: It's hard to believe we'd ever get here as this jar provides the  :smile: 

Probably ok but better covered with a unit test.

File: geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<head>
+<title>Composer API Configuration</title>
+<meta name="ROBOTS" content="NOINDEX, NOFOLLOW"/>
+</head>
+<body>
+<h2>Composer API Configuration</h2>
+<#if model["requests"]?size != 0>
+<ul>
+<#foreach request in model["requests"]>
+<li>${request["method"]} <a href="${model["baseUrl"]}${request["pattern"]}">${request["pattern"]}</a></li>
```
Comment (ischneider) [inline geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl]: if I click the link, it fails (and I understand why) so perhaps no link for now?

**Comment by ischneider** [comment]
Minor inline code comments.

File: geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl (hunk 2 above)
Comment (tbarsballe) [inline geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl]: Any of the links that don't have wildcards in their path work fine (eg. api/formats, api/stores, etc.)

File: geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl (hunk 1 above)
Comment (tbarsballe) [inline geoserver/webapp/src/main/webapp/WEB-INF/ftl/api.ftl]: Fair. That bit is mostly leftover from the GeoServer REST API page this is based on (Which had a similar situation, in that it also provided API extensions)

**Pull request merged**